enterprise/kernel/src/main/java/org/neo4j/kernel/impl/pagecache/PageCacheWarmer.java

```diff
@@ -0,0 +1,324 @@
+        finally
+        {
+            IOUtils.closeAll( files );
+        }
+        pageCache.reportEvents();
+        return stopped ? OptionalLong.empty() : OptionalLong.of( pagesInMemory );
+    }
+
+    private long profile( PagedFile file ) throws IOException
+    {
+        long pagesInMemory = 0;
+        File outputNext = profileOutputFileNext( file );
+
+        try ( OutputStream outputStream = compressedOutputStream( outputNext );
+              PageCursor cursor = file.io( 0, PF_SHARED_READ_LOCK | PF_NO_FAULT ) )
+        {
+            int stepper = 7;
+            int b = 0;
+            for ( ; ; )
+            {
+                if ( stopped )
+                {
+                    return pagesInMemory;
+                }
+                if ( !cursor.next() )
+                {
+                    break; // Exit the loop if there are no more pages.
+                }
+                b >>= 1;
+                if ( cursor.getCurrentPageId() != PageCursor.UNBOUND_PAGE_ID )
+                {
+                    pagesInMemory++;
+                    b += 0b1000_0000;
+                }
+                if ( stepper == 0 )
+                {
+                    outputStream.write( b );
+                    b = 0;
+                    stepper = 7;
+                }
+                else
+                {
+                    stepper--;
+                }
+            }
+            b >>= stepper + 1;
+            outputStream.write( b );
+            outputStream.flush();
+        }
+
+        File outputFinal = profileOutputFileFinal( file );
+        fs.renameFile( outputNext, outputFinal, StandardCopyOption.REPLACE_EXISTING );
+        return pagesInMemory;
+    }
+
+    private InputStream compressedInputStream( File input ) throws IOException
+    {
+        InputStream source = fs.openAsInputStream( input );
+        try
+        {
+            return compressors().createCompressorInputStream( COMPRESSION_FORMAT, source );
+        }
+        catch ( CompressorException e )
+        {
+            IOUtils.closeAllSilently( source );
+            throw new IOException( "Exception when building decompressor.", e );
+        }
+    }
+
+    private OutputStream compressedOutputStream( File output ) throws IOException
+    {
+        OutputStream sink = fs.openAsOutputStream( output, false );
+        try
+        {
+            return compressors().createCompressorOutputStream( COMPRESSION_FORMAT, sink );
+        }
+        catch ( CompressorException e )
+        {
+            IOUtils.closeAllSilently( sink );
+            throw new IOException( "Exception when building compressor.", e );
+        }
+    }
+
+    private CompressorStreamFactory compressors()
+    {
+        return new CompressorStreamFactory( true, 1024 );
+    }
+
+    private File profileOutputFileFinal( PagedFile file )
+    {
+        File mappedFile = file.file();
+        String profileOutputName = "." + mappedFile.getName() + SUFFIX_CACHEPROF;
```

Comment:
I don't think people want to see these by default since they're not super important. I hope hiding them means we'll get fewer support cases that are just about asking what these files are.